libsolidity/interface/Natspec.cpp

```diff
@@ -100,15 +100,52 @@ Json::Value Natspec::devDocumentation(ContractDefinition const& _contractDef)
 		{
 			Json::Value method(devDocumentation(fun->annotation().docTags));
 			if (!method.empty())
+			{
 				// add the function, only if we have any documentation to add
+				Json::Value ret(Json::objectValue);
+
+				auto returnParams = fun->returnParameters();
+				auto returnDoc = fun->annotation().docTags.equal_range("return");
+
+				if (!returnParams.empty())
+				{
+					unsigned int n = 0;
+					for (auto i = returnDoc.first; i != returnDoc.second; i++)
+					{
+						string paramName = returnParams.at(n)->name();
+						string content = i->second.content;
+
+						if (paramName.empty())
+						{
+							paramName = "_" + std::to_string(n+1);
+						}
+						else
+						{
+							//check to make sure the first word of the doc str is the same as the return name
+							auto nameEndPos = content.find_first_of(" \t");
+							solAssert(content.substr(0, nameEndPos) == paramName, "No return param name given: " + paramName);
+							content = content.substr(nameEndPos+1);
+						}
```

Comment:
@erak I know I suggested earlier to move this work into the `docParser`, but since it only keeps track of/iterates over the strings associated with the notices it doesn't know whether or not that return parameter is named via the AST and thus should enforce the error.  

Hope this solution isn't too hacky...